osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmap.cs

```diff
@@ -115,25 +124,62 @@ private void load(SongSelect songSelect, BeatmapManager manager, BeatmapSetOverl
                                         },
                                         new OsuSpriteText
                                         {
-                                            Text = $"{(beatmap.Metadata ?? beatmap.BeatmapSet.Metadata).Author.Username}",
+                                            Text = $"{(Beatmap.Metadata ?? Beatmap.BeatmapSet.Metadata).Author.Username}",
                                             Font = OsuFont.GetFont(italics: true),
                                             Anchor = Anchor.BottomLeft,
                                             Origin = Anchor.BottomLeft
                                         },
                                     }
                                 },
-                                starCounter = new StarCounter
+                                starCounterContainer = new FillFlowContainer
                                 {
-                                    CountStars = (float)beatmap.StarDifficulty,
-                                    Scale = new Vector2(0.8f),
+                                    Direction = FillDirection.Horizontal,
+                                    Spacing = new Vector2(4, 0),
+                                    AutoSizeAxes = Axes.Both,
+                                    Children = new Drawable[]
+                                    {
+                                        starCounter = new StarCounter
+                                        {
+                                            CountStars = (float)Beatmap.StarDifficulty,
+                                            Scale = new Vector2(0.8f),
+                                        }
+                                    }
                                 }
                             }
                         }
                     }
                 }
             };
+
+            // initialize top rank
+            if (TopRank.Value.HasValue)
+                loadTopRank(TopRank.Value);
         }
 
+        private void loadTopRank(ScoreRank? score)
+        {
+            if (!score.HasValue && rank != null)
+            {
+                starCounterContainer.Remove(rank);
+                rank = null;
+            }
+            else if (rank == null && starCounterContainer != null && score.HasValue)
+            {
+                starCounterContainer.Insert(-1, rank = new UpdateableRank(score.Value)
+                {
+                    RelativeSizeAxes = Axes.None,
+                    Size = new Vector2(40, 20),
+                    Scale = new Vector2(0.8f)
+                });
+            }
+            else if (rank != null && score.HasValue)
+            {
+                rank.Rank = score.Value;
+            }
```

Comment:
Now that I look at this logic I think `UpdateableRank` should just accept a null rank and not draw anything if it is passed a null. This is too much.